### ISSUE TITLE
draw.list file

### ISSUE THREAD

**Issue by maliky** [maths/moodle-qtype_stack#149]
I've update Moodle and stack v2015090900

I then encounter a problem with stack or maxima?

Copying the health script code in maxima v5.34.1, which seem to run find on my machine I get this error

```
load(draw.lisp);

Maxima encountered a Lisp error:

 Condition in MACSYMA-TOP-LEVEL [or a callee]: INTERNAL-SIMPLE-ERROR: ((MNCTIMES
                                                                        SIMP)
                                                                       $DRAW
                                                                       $LISP) cannot be coerced to a namestring.
```
Any idea what could have been causing this ?

Before that I was able to use stack and maxima but the answer previsualisation was not working too good so I had desactivated it.

**Comment by sangwinc** [comment]
This is a problem which has surfaced before.  I think you don't have a fully installed version of Maxima.  Can you provide me more information about the Maxima version?  E.g. which version, which lisp, and did you compile it from source?

Chris

**Comment by maliky** [comment]
sorry for the delay but this issue was not closed for me.  I still cannot
answer about lisp version but I do not compile from source.  It will try
that solution as soon as possible.

I installed maxima from jessie repositories but my linux kernel is old,
3.16 because I cant upgrade it as i am running in an openvz vps.

please keep it open for 2 more weeks.

thank you

Malik Koné,
envoyé depuis Abidjan avec un mobile

**Comment by maliky** [comment]
a 
#dpkg -l | grep lisp 
returns nothing on my machine and when I do 
#apropos lisp
forwad me to maxima.

Maxima used to work on my machine. It stoped working I don't really know when.  After a 
#apt-get upgrade 
I gess.

I going to remove it and install it from source.
I'm getting back to tell you how it went.

**Comment by sangwinc** [comment]
Ok, please do let me know how you get on.  I'd like to document the symptons and solution for anyone else who encounters this problem.

Chris

**Comment by maliky** [comment]
#apt-get install gcl 
#export GCL_ANSI=t
#apt-get install build-essential automake texinfo
#git clone git://git.code.sf.net/p/maxima/code maxima-code
#cd maxima-code 
#sh ./bootstrap
#./configure --enable-gcl
#LANG=C make
#LANG=C make install
#maxima
>run_testsuite();

I now have maxima 5-37 dans gcl 2.6.12 up and running but still the health script see an error...
Here the CAS result

Maxima branch_5_37_base_100_g53313a6 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) 
loadfile: failed to load /usr/local/share/maxima/branch_5_37_base_100_g53313a6/share/draw/draw.lisp

...
But the file draw.lisp is there.

**Comment by maliky** [comment]
Here the content of my draw file.
I don't know how to debbug it.

(in-package #-gcl #:maxima #+gcl "MAXIMA")
($put '$draw 2 '$version)
#+ecl ($load "lisp-utils/defsystem.lisp")
(load (merge-pathnames (make-pathname :name "draw" :type "system")
                       #-gcl *load-pathname*
                       #+gcl sys:*load-pathname*))
(mk:oos "draw" :compile)

**Comment by aharjula** [comment]
Some of the 5.37.x versions of Maxima have had problems with some other things so maybe you should try something that we know works. I would recommend 5.35.1. Anyway, my list of sanity checks:

 1. Does your system provide maxima as an package, can you use that or is it too old?
 2. GCL may work but CLISP works better in my systems.
 3. When running configure before compiling does it say anything about missing stuff?
 4. Do you have these packages installed as dev versions that you can compile against:
   - texinfo is a package often missing
   - recode is another
 5. When running maxima might want to write cached pre-compiled versions of various scripts to the home directory of the running user. Maybe to ~/.maxima can it do that? Does the user running it even have a home directory?
 6. You said that you installed maxima from jessie-repos, is the VPS you are running on even a Debian and has it other stuff that might have been mixed from different releases?
 7. If the file /usr/local/share/maxima/branch_5_37_base_100_g53313a6/share/draw/draw.lisp is there is it readable? Can the user running maxima read it and all the directories (execute right for directories) up to it.
 8. That path /usr/local/share/maxima/branch_5_37_base_100_g53313a6 sure looks odd are you sure you configured and installed it correctly?

**Comment by aharjula** [comment]
Do post the output of the configure script and the maximalocal.mac so that we have something to wonder about.

**Comment by maliky** [comment]
Aharjula,

Thank you for reopening.
1. Does your system provide maxima as an package, can you use that or is it too old?
Yes it does. I use maxima on command line without problem.

2. GCL may work but CLISP works better in my systems.
I'm runing debian jeesie with an old kernel because of virtualisation.  I'll try clisp if every thing else fails.

3. When running configure before compiling does it say anything about missing stuff?
Yes I had to install texinfo for exemple. I could not compile gcl from source and use the apt-get instead with an option to take care of ANSI. (see above for details of command lines used)

4. Do you have these packages installed as dev versions that you can compile against:
texinfo is a package often missing
recode is another
I don't know.  I don't think they are. I used a package manager to install texinfo. 
Recode was not installed. now it is. 

5. When running maxima might want to write cached pre-compiled versions of various scripts to the home directory of the running user. Maybe to ~/.maxima can it do that? Does the user running it even have a home directory?
I don't know what user is running maxima when I use it with stack.  Is it www-data ? if so, no, my apache serveur does not have a home dir.

You said that you installed maxima from jessie-repos, is the VPS you are running on even a Debian and has it other stuff that might have been mixed from different releases?
The VPS is using OpenVZ and run Jeesie packages with an old linux kernel.  (2.6.32)  I cannot avoid that and it did not cause a probleme a few month ago.  Would I really have to change host ?

If the file /usr/local/share/maxima/branch_5_37_base_100_g53313a6/share/draw/draw.lisp is there is it readable? Can the user running maxima read it and all the directories (execute right for directories) up to it.
Well yes.  I have a loading problem for that file even when the root is runngin maxima.  I have checked the path to the file. the root can see it and open it from outside maxima.

That path /usr/local/share/maxima/branch_5_37_base_100_g53313a6 sure looks odd are you sure you configured and installed it correctly?
Well no.. I will reinstall it more carefully.  I'm not a developper myself and I not used to compile install stuff.  I didn't know the path looked wired so that's a good way for me to checked that I probably had a problem during config or install.  
Thank you.

**Comment by maliky** [comment]
Here my maxima configuration file.

```
/* ***********************************************************************/
/* This file is automatically generated at installation time.            */
/* The purpose is to transfer configuration settings to Maxima.          */
/* Hence, you should not edit this file.  Edit your configuration.       */
/* This file is regularly overwritten, so your changes will be lost.     */
/* ***********************************************************************/

/* File generated on October 30, 2015, 2:17 pm */

/* Add the location to Maxima's search path */
file_search_maxima:append( [sconcat("/var/www/html/edu/question/type/stack/stack/maxima/###.{mac,mc}")] , file_search_maxima)$
file_search_lisp:append( [sconcat("/var/www/html/edu/question/type/stack/stack/maxima/###.{lisp}")] , file_search_lisp)$
file_search_maxima:append( [sconcat("/var/www/moodledata/stack/logs/###.{mac,mc}")] , file_search_maxima)$
file_search_lisp:append( [sconcat("/var/www/moodledata/stack/logs/###.{lisp}")] , file_search_lisp)$

STACK_SETUP(ex):=block(
    MAXIMA_VERSION_NUM_EXPECTED:36.1,
    MAXIMA_PLATFORM:"unix",
    maxima_tempdir:"/var/www/moodledata/stack/tmp/",
    IMAGE_DIR:"/var/www/moodledata/stack/plots/",
    PLOT_TERMINAL:"png",
    PLOT_TERM_OPT:"large transparent size 450,300",
    DEL_CMD:"rm",
    GNUPLOT_CMD:"/usr/bin/gnuplot",
    MAXIMA_VERSION_EXPECTED:"5.36.1",
    URL_BASE:"!ploturl!",
    true)$
/* Load the main libraries */
load("stackmaxima.mac")$
```

Here the config.log for maxima.  http://pastebin.com/6uNXxnRE

**Comment by aharjula** [comment]
1. Which version of maxima does the distribution provide? Is it really too old?
2. Kernel version does not matter what matters is that you are running a consistent linux distribution i.e. you have not mixed packages from multiple different distributions and built a mess where you could not even find the libraries if they were there. So you have a Debian jessie and you have used delian-jessie packages on it? ``lsb_release -a`` would tell us a lot about that, if not installed on Debian the package name is "lsb-release".
3. You do not need to compile GCL or CLISP from source the versions provided by your distribution should be new enough they should not be over five years old. Just install them from apt-get.
5. The user running your HTTP-server will be the user running the PHP-code that calls maxima, with apache and recent Debians that user is indeed "www-data", and typically its home directory is /var/www but it does not have the rights to write to it. You may try creating that .maxima directory ``mkdir ~www-data/.maxima; chown www-data:www-data ~www-data/.maxima; chmod 770 ~www-data/.maxima`` but once things work remove the write permissions from that directory and everything under it.
6. OpenVZ is a container system that could be running on top of basically anything and the containers in it can be basically anything, do run the ``lsb_release -a`` command and give its output so we can be sure that our obvious language barriers are not hiding the real situation. As container virtualisation systems are typically used to contain certain functions they may deny some actions but I would be very surprised if that is the case here, or have you been turning on random security features without understanding what they do? 
7&8. Well if the root can't read it then something is obviously wrong, the file must be in the wrong place. Did your installation process go like this:
```
# Get the source package.tar.gz place it to a temp-location
cd /tmp/
# extract the package
tar xzf package.tar.gz
# go to the folder extracted from it
cd maxima-5.xx.y
# do the configure thing, maybe with some extra parameters probably --prefix=/opt/maxima/5.xx.y
./configure
# then compile, takes a couple of minutes, should not give errors if it does then you have a problem
make
# then check, this is optional
make check
# and finally install
sudo make install
# then run it from the installation target which if you used that prefix would be this
/opt/maxima/5.xx.y/bin/maxima
```
Basically you installed with "make install" and not by copying stuff by hand?

As to the files you gave according to the maximalocal.mac the maxima-scripts of STACK are in /var/www/html/edu/question/type/stack/stack/maxima/ and it would probably be safe to assume that the user running maxima can read them. The config.log says nothing obvious but that is the raw form of it and I would rather see the form outputted by the configure command itself. Please run configure with the parameters that you used and save its output to a file and post that. ``./configure --??? > the-file.log``.

**Comment by maliky** [comment]
1.  The maxima version of debian is not old at all it's the 5.34.1-2.  So I never understood why at some point stack complained about it.

2. ```
#lsb_release -a
Distributor ID: Debian
Description:    Debian GNU/Linux 8.2 (jessie)
Release:        8.2
Codename:       jessie
```
I don't recall messing up with my sources.  I installed everything with apt-get until something fails.  That's when I try to comply from sources.
 I run apt-get update and apt-get upgrade regularly.

3. I installed gcl from repositories.  I gave the headers further up.  
4.  Done
5.  I did install a mail server packaged called iRedMail using a script that settup https.  but stack worked for several month after the installation.  I did not install something recently that could mess the security features.
6. Yes I followed classic install installation procedure without copying stuff by hand.
first I tryed reinstalling maxima with apt-get since it did not solve my problem I went for the git source.
As describe previously here's the installation procedure I used.

```
#apt-get install gcl 
#export GCL_ANSI=t
#apt-get install build-essential automake texinfo
#git clone git://git.code.sf.net/p/maxima/code maxima-code
#cd maxima-code 
#sh ./bootstrap
#./configure --enable-gcl
#LANG=C make
#LANG=C make install
#ln -s /usr/local/bin/maxima /usr/bin/
#maxima
```
then in maxima
```
run_testsuite();
```
No problems were detected with the tests.
And sure www-data can read all files under /var/www/html/ and so also for /var/www/html/edu/question/type/stack/stack/maxima/


What I find strange is that running maxima as root, I can load("/var/www/moodledata/stack/maximalocal.mac"); but then I have an error when the script wants to read the draw.lisp file.
and I have a load error even when 
```cd  /usr/local/share/maxima/branch_5_37_base_100_g53313a6/share/draw/```
run maxima  as root and
```(%i1) load("draw.lisp);```
Also I can load other lisp files in that directory.

**Comment by aharjula** [comment]
i am slightly concerned about using the development branch of maxima instead of a release, who knows what special changes it might contain, maybe try downloading or cloning some specific release. In addition to that I find it odd that if you do not define the installation prefix for it why does it place it to that path, maybe it uses the branch/commit name as an version or something.

Lately STACK has been checking if the version of the Maxima matches the version defined in STACKs settings and if your version is "branch_5_37_base_100_g53313a6" and it expects "5.36.1" as your maximalocal.mac (comes from Moodle side settings) says that might cause issues later but again this has nothing to do with the draw.lisp issue.

When you say that you reinstalled maxima from the package system how did you do it? Did you first remove the old or did you purge the old and then install or did you just try to upgrade it by installing on top of the old?

You have gnuplot installed? And you can plot with it, from maxima?

Anyway I would still want to see that output from configure but I am starting to think that this is an issue with Maxima and some Debian core-library.

**Comment by maliky** [comment]
yes I purge maxima and reinstalled
̀̀̀̀̀̀̀́́́́dpkg -p maxima*
apt-get install maxima

I ll do a reinstall today and share my config.

sorry to bother you with what also looks like a maxima issue ti me.  but
things worked not to bad a 2 months ago.
the stack  health check was ok.  i could draw.  the only issue then was
with immediate feed back.   I mean that student could not see the formated
result of their answer as they typed it.  so I unchecked that option.

thank you

Malik Koné,
envoyé depuis Abidjan avec un mobile

**Comment by maliky** [comment]
I recompiled maxima,
same issue.

A few years ago I remember having a problem sith coercing path strings of files to load but today
Even starting maxima with root in the draw.lisp directory I'm not able to load draw.lisp but I can load other lisp files from the same directory

Here's my config file from configure script.  http://pastebin.com/6uNXxnRE

---
next step will be to reinstall from repositories...  
I really don't know what to do.

maxima seems to work find.

Only problem is to load the draw.lisp file even if I'm root and in the same directory.

Maybe you can guide me to clone a better (more stable) maxima than the one I did.   
what adresse should I clone?
what branch should I switch too ?

Thank you

**Comment by sangwinc** [comment]
Ok, I'm really sorry you are having such problems.  I've had an email from another user with the same issue.  I've spent some time this afternoon installing Debian Jessie on an old machine from scratch and doing a completely clean, bottom up install of *everything* from distribution packages. (apt-get install etc.)  It is helpful for developers to do this once in a while!

Debian currently is shipped with
Maxima 5.34.1 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12 (a.k.a. GCL)

To minimally reproduce your problem type in
load("stats");
into the maxima command line.

The statistics packages are optional, but they are clearly breaking the rest of STACK.   I've taken the rather drastic step of removing them for now.  See https://github.com/maths/moodle-qtype_stack/commit/75f4ea0abcd0bde46ead6da3f978b0fd5269f5a2

Please update the qtype_stack code to the latest in the master branch.  I was right about the problem being in maxima (not in STACK) but that is very unhelpful! 

I'll give some thought about how to sort out an option for those people relying on the stats packages in older systems.  This is very annoying!   I hope this gets STACK working for you.

**Comment by maliky** [comment]
Matti Merci beaucoup.  ****It works !**** 


I've sent an email to maxima mailling list.  I hope they can do something quick.

**Comment by sangwinc** [comment]
Thank you very much for reporting this and I'm sorry I didn't attend to it sooner.  I've been very busy teaching myself over the last couple of weeks.... Would you like to close the issue?

**Comment by maliky** [comment]
ok but please tell me what s the issue then with the draw or stat file so i
can discuss it with maxima people.

thank you

Malik Koné,
envoyé depuis Abidjan avec un mobile

**Comment by aharjula** [comment]
Ok. So Chris confirmed that there is an issue with Maxima on Debian and the simplest way to get that issue to manifest itself is to load the stats package in Maxima. I am still intending to try this on the newer Ubuntus but things work on the last LTS-version. If its something that appears in Debian but not in Ubuntu then we might get more information but otherwise this is something that is not really something that we can use too much time with.

Instead of using git to get the correct version you might want to do things the old fashioned way and just download the code in the form they have packaged it as they may have dome some things to that code and added some details that do not appear in the git-repository, you can get such packages from here:  http://sourceforge.net/projects/maxima/files/Maxima-source/

As to trying to get the correct version from git. After you have cloned that maxima-code as you have done already you can simply switch to a tagged version of it:

```
git clone git://git.code.sf.net/p/maxima/code maxima-code
cd maxima-code/
# Then list the tags available
git tag -l
# and switch to a specific one like the 5.35.1
git checkout tags/5.35.1
# and do regenerate the configure script with the bootstrap (remove the configure script first) after switching the version/tag
```

Thank you for the output of the configure script it says nothing about any missing things so maybe Maxima does not check for the thing that now is broken or maybe Debian has broken that thing in some interesting way so that the configure script believes it to exist but does not.

**Comment by aharjula** [comment]
Something interesting just happened. I have been installing various versions of Ubuntu to my test farm to see if this affects Ubuntu also as that is my target audience. When I installed maxima from package manager to vivid and started it it said this:

```
root@vivid-test:/root# maxima

Maxima 5.35.1 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12 (a.k.a. GCL)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) load("stats");

Compiling /usr/share/maxima/5.35.1/share/draw/grcommon.lisp.
End of Pass 1.  
End of Pass 2.  
sh: 1: gcc: not found
loadfile: failed to load /usr/share/maxima/5.35.1/share/draw/draw.lisp
 -- an error. To debug this try: debugmode(true);
```

And as it says "gcc not found" I installed gcc for it by installing build-essential and when I tried again it compiled some new files and finally loaded that file:

```
root@vivid-test:/root# maxima

Maxima 5.35.1 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12 (a.k.a. GCL)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) load("stats");

Compiling /usr/share/maxima/5.35.1/share/draw/grcommon.lisp.
End of Pass 1.  
End of Pass 2.  
OPTIMIZE levels: Safety=2, Space=3, Speed=3
Finished compiling /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/grcommon.o.
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/grcommon.o
start address -T 0x2547560 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/grcommon.o
Compiling /usr/share/maxima/5.35.1/share/draw/gnuplot.lisp.
End of Pass 1.  

;; Note: Tail-recursive call of BIPART was replaced by iteration.
;; Note: Tail-recursive call of BIPART was replaced by iteration.
End of Pass 2.  
OPTIMIZE levels: Safety=2, Space=3, Speed=3
Finished compiling /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/gnuplot.o.
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/gnuplot.o
start address -T 0x2561ec0 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/gnuplot.o
Compiling /usr/share/maxima/5.35.1/share/draw/vtk.lisp.
End of Pass 1.  
End of Pass 2.  
OPTIMIZE levels: Safety=2, Space=3, Speed=3
Finished compiling /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/vtk.o.
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/vtk.o
start address -T 0x25a4fa0 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/vtk.o
Compiling /usr/share/maxima/5.35.1/share/draw/picture.lisp.
End of Pass 1.  
End of Pass 2.  
OPTIMIZE levels: Safety=2, Space=3, Speed=3
Finished compiling /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/picture.o.
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/picture.o
start address -T 0x25cb1e0 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/picture.o
(%o1)           /usr/share/maxima/5.35.1/share/stats/stats.mac
root@vivid-test:/root# maxima

Maxima 5.35.1 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12 (a.k.a. GCL)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) load("stats");

Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/grcommon.o
start address -T 0x2547560 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/grcommon.o
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/gnuplot.o
start address -T 0x255fae0 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/gnuplot.o
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/vtk.o
start address -T 0x25a4c50 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/vtk.o
Loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/picture.o
start address -T 0x25cae90 Finished loading /root/.maxima/binary/5_35_1/gcl/GCL_2_6_12/share/draw/picture.o
(%o1)           /usr/share/maxima/5.35.1/share/stats/stats.mac

```
You may wish to check that you have gcc installed and maybe the part of that error that would tell you about that is hidden in your version or there is some other reason why this little detail was hidden. But obviously even the Ubuntu package form Maxima seems to be broken if it does not require gcc...

But I now I have a problem... for some reason I can't compile those git-clones of the code correctly they just don't work and generate wrong paths but when I use the same configure settings for those packaged versions everything works. Tried to script the building of all the maxima versions supported by STACK and things have not gone well.

**Comment by aharjula** [comment]
And maybe you should not load "draw.lisp" maybe you should load "draw" as if it contains some compile logic it might need to be loaded through some other logic...

**Comment by aharjula** [comment]
Additional information about Ubuntus, the current Xenial dev version has 5.37.2 and it says this:
```
root@xenial-test:/root# maxima

Maxima 5.37.2 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) load("stats");

loadfile: failed to load /usr/share/maxima/5.37.2/share/draw/draw.lisp
 -- an error. To debug this try: debugmode(true);
```

Wily at least gives that gcc clue:

```
root@wily-test:~# maxima

Maxima 5.36.1 http://maxima.sourceforge.net
using Lisp GNU Common Lisp (GCL) GCL 2.6.12
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) load("stats");

Compiling /usr/share/maxima/5.36.1/share/draw/grcommon.lisp.
End of Pass 1.  
End of Pass 2.  
sh: 1: gcc: not found
loadfile: failed to load /usr/share/maxima/5.36.1/share/draw/draw.lisp
 -- an error. To debug this try: debugmode(true);
(%i2) 
```

Wily works after installation of "build-essential" and but xenial did not work, maybe due to being broken otherwise, being in development still. Or maybe that 5.37.x has even more problems that we didn't even know about.

Same happened with a fresh trusty, but it was fixable a mere "gcc" was enough and I would imagine it would be enough on the others too.

**Comment by sangwinc** [comment]
The problems with 5.37.x are mathematical changes in behavior which break some of the subtle answer tests.  I can see why the Maxima people didn't notice them, but sometimes this is exactly what we are testing, e.g. commutative of real multiplication!

These problems are installation and loading of draw.  They are completely separate.  I'll email the Maxima list and ask about this.  I expect it has been raised before....

**Comment by maliky** [comment]
thank you

Malik Koné,
envoyé depuis Abidjan avec un mobile

**Comment by aharjula** [comment]
>These problems are installation and loading of draw. They are completely separate. I'll email the Maxima list and ask about this. I expect it has been raised before....

More details from the Ubuntu testing front:
 - I tried compiling most of the Maxima versions supported by STACK using fairly normal build procedures "./configure --prefix;make;sudo make install".
 - I then tested those versions by pushing 'load("stats");quit();' to them and checking if that returns anything sensible.
 - The results were not good for older versions but newer ones seem to work, probably there is some issues with the compilation process or with newer versions of libraries present.
 - Repeated that on those fresh installs of Ubuntu trusty, vivid and wily by running these two scripts:
```
#!/bin/sh
## maxima-build.sh
# This script builds various versions of maxima for test purposes.

versions="5.26.0 5.27.0 5.28.0 5.30.0 5.31.1 5.31.2 5.31.3 5.32.0 5.32.1 5.33.0 5.34.0 5.34.1 5.35.1 5.36.0 5.36.1"
target="/opt/maxima-test/"  # the --prefix will be $target$version

sudo aptitude build-dep maxima
sudo aptitude install maxima git build-essential clisp texinfo automake librecode-dev gnuplot maxima-share

sudo mkdir -p $target

if [ ! -e "maxima-code" ]; then
git clone git://git.code.sf.net/p/maxima/code maxima-code
fi
cd maxima-code/
git pull

for version in $versions; do
 # Ensure that everything has been cleaned...
 rm -Rf *
 git checkout .
 git checkout tags/$version
 ./bootstrap
 ./configure --prefix=$target$version
 LANG=C make
 LANG=C make install
done
```
```
#!/bin/sh
## maxima-test.sh
# This script tests various versions of maxima, use the maxima-build.sh to create the testing environment.

versions="5.26.0 5.27.0 5.28.0 5.30.0 5.31.1 5.31.2 5.31.3 5.32.0 5.32.1 5.33.0 5.34.0 5.34.1 5.35.1 5.36.0 5.36.1"
target="/opt/maxima-test/"  # the --prefix will be $target$version

for version in $versions; do
 # Ensure that caches have been cleaned...
 # the earlier versions of maxima may use the same names for cache files and that may cause trouble
 rm -Rf ~/.maxima
 echo 'load("stats");quit();' | $target$version/bin/maxima
 echo 'load("stats");quit();' | $target$version/bin/maxima
done
```

The results are:
 trusty: 5.26.0 - 5.30.0 and 5.34.1 do not load("stats")
 vivid: 5.26.0 - 5.30.0 do not load("stats")
 wily: 5.26.0 - 5.30.0 do not load("stats")

I have no idea why 5.34.1 does not work on trusty. Others seem to work and it seems that from version 5.35.1 forward the naming convention of the cached files can handle situations where you have multiple version of maxima present.

**Comment by robert-dodier** [comment]
I believe the failure to load draw.lisp (and therefore the failure to load stats) is due to commit 367d0e22 on Maxima Git. I've replied to M. Kone's message on the Maxima mailing list and I'll report a resolution here when we have one.

**Comment by robert-dodier** [comment]
My advice about working around this bug is to use SBCL or Clisp or something else instead of GCL.

**Comment by maliky** [comment]
@robert-dodier Thank you.

**Comment by robert-dodier** [comment]
I believe this bug is fixed by commit 623a513 on Maxima Git master. I have merged it into the 5.37 release branch and I'll make another 5.37 release which contains it. (Yes, I know I said that about another bug fix. I'll really do it this time, and include that other bug fix -- something about commutativity of addition or lack of it -- also.)

**Comment by sangwinc** [comment]
Thank you Robert, we do appreciate your work here.
Chris

**Comment by robert-dodier** [comment]
I have merged 623a513 into branch-5_37 and also merged a patch for commutativity (as Chris and I discussed back in September -- sorry for the delay). I have tagged 5.37.3 and uploaded tar.gz and rpms to Sourceforge. Maliky, I believe you should be able to build 5.37.3 with either GCL <= 2.6.12 or GCL > 2.6.12 -- should work either way, if it doesn't, well, please let me know. Hope this helps & please keep me posted. -- Robert.

**Comment by robert-dodier** [comment]
PS. Sourceforge file manager for Maxima is: http://sourceforge.net/p/maxima/files

**Comment by sangwinc** [comment]
Thanks Robert,
I've complied Maxima 5.37.3 with no problems, and this now loads the stats packages etc. no problem.

We have one failing unit test as follows which is a change.   A minimal example is below.

The old version is
Maxima 5.36.1 http://maxima.sourceforge.net
using Lisp CLISP 2.49 (2010-07-07)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) display2d:false;
(%o1) false
(%i2) simp:false;
(%o2) false
(%i3) (3/6)*x;
(%o3) (3/6)*x
(%i4) 

The new version gives

Maxima 5.37.3 http://maxima.sourceforge.net
using Lisp CLISP 2.49 (2010-07-07)
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) display2d:false;
(%o1) false
(%i2) simp:false;
(%o2) false
(%i3) (3/6)*x;
(%o3) (3*x)/6


So, with simp:false we have a change of behaviour of
(3/6)*x -> (3*x)/6
This matters because I want to filter out coefficients in polynomials and rational numbers in other expressions and decide if a student has entered numbers in "lowest terms".  I know this is a very subtle change, but it breaks the "lowest terms" test.  Can anything be done about this?

**Comment by robert-dodier** [comment]
Thanks for the info, Chris. Yes, I can revise the output again. I'll try harder to make the current output match the old output.

Can I ask you to open a separate issue about that? If you just copy your comment about `(3/6)*x` to a new issue, that would be great. It helps me keep my head straight, thanks for indulging me.

**Comment by sangwinc** [comment]
Thanks Robert,
https://sourceforge.net/p/maxima/bugs/3046/
